imswitch/imcommon/model/dirtools.py

```diff
@@ -1,90 +1,149 @@
-import glob
 import os
+import platform
 from abc import ABC
 from pathlib import Path
 from shutil import copy2
 
 import imswitch
 
+from xdg import xdg_config_home, xdg_data_home
 
-def getSystemUserDir():
-    """ Returns the user's documents folder if they are using a Windows system,
-    or their home folder if they are using another operating system. """
 
-    if os.name == 'nt':  # Windows system, try to return documents directory
-        try:
-            import ctypes.wintypes
-            CSIDL_PERSONAL = 5  # Documents
-            SHGFP_TYPE_CURRENT = 0  # Current value
+def _getWindowsDir():
+    try:
+        import ctypes.wintypes
+        CSIDL_PERSONAL = 5  # Documents
+        SHGFP_TYPE_CURRENT = 0  # Current value
 
-            buf = ctypes.create_unicode_buffer(ctypes.wintypes.MAX_PATH)
-            ctypes.windll.shell32.SHGetFolderPathW(0, CSIDL_PERSONAL, 0, SHGFP_TYPE_CURRENT, buf)
+        buf = ctypes.create_unicode_buffer(ctypes.wintypes.MAX_PATH)
+        ctypes.windll.shell32.SHGetFolderPathW(
+            0, CSIDL_PERSONAL, 0, SHGFP_TYPE_CURRENT, buf)
 
-            return buf.value
-        except ImportError:
-            pass
-        #TOOD: How can we ensure that configuration files are updated automatically.. 
-    return os.path.expanduser('~')  # Non-Windows system, return home directory
+        return Path(buf.value)
+    except ImportError:
+        pass
+    # TOOD: How can we ensure that configuration files are updated automatically..
 
 
-_baseDataFilesDir = os.path.join(os.path.dirname(os.path.realpath(imswitch.__file__)), '_data')
-_baseUserFilesDir = os.path.join(getSystemUserDir(), 'ImSwitchConfig')
+def _getImSwitchDir(config: bool = False) -> str:
+    """Returns an XDG compatible data/config folder on Linux, the user's
+    documents folder on Windows or their home folder if they are using another
+    operating system.
 
+    """
+    if platform.system() == 'Linux':
+        # return XDG compatible directory,
+        # see https://www.freedesktop.org/wiki/Specifications/basedir-spec/
+        if config:
+            return xdg_config_home() / 'ImSwitch'
+        else:
+            return xdg_data_home() / 'ImSwitch'
 
-def initUserFilesIfNeeded():
-    """ Initializes all directories that will be used to store user data and
-    copies example files. """
+    else:
+        # on any other OS put data and configs into same folder
+        if platform.system() == 'Windows':  # Windows system, try to return documents directory
+            # TODO why would these files not go into .AppData\Roaming?
```

Comment:
I personally prefer to keep these files into "Documents" because it's easier to access - especially if somebody wants to make a new configuration file for a new setup. At some point we will have to add a configuration manager to be able to build up your own setup from the GUI but right now it's best to keep the config folder in an easier spot to reach.